lib/CallBackery/GuiPlugin/AbstractCardlist.pm

```diff
@@ -78,108 +78,41 @@ Configuration of the card list cards
 =cut
 
 has cardCfg => sub {
-    die mkerror(3456, trm("cardCfg must be defined in child class"));
+    die mkerror(3456, trm("cardCfg must be defined in child plugin class"));
 };
 
-=head2 screenCfg
+=head2 tableCfg
 
-Custom default screen configuration.
+is not used for card plugins, use cardCfg instead.
 
 =cut
 
-has screenCfg => sub {
-    my $self = shift;
-    my $screen = $self->SUPER::screenCfg;
-    $screen->{type}    = 'cardlist';
-    $screen->{cardCfg} = $self->cardCfg;
-    return $screen;
-};
-
-=head2 Auto refresh action
-
-Add something like the following to your derived plugins to get an automatic
-refresh of the CardList.
-
- has actionCfg => sub {
-     my $self = shift;
-     return [{
-         action   => 'refresh',
-         interval => $self->refreshInterval,
-     }];
- };
-
- has refreshInterval => sub {
-     return 1000; # in milliseconds
- };
+has 'tableCfg';
 
 =head1 METHODS
 
-All the methods of L<CallBackery::GuiPlugin::AbstractForm> plus:
+All the methods of L<CallBackery::GuiPlugin::AbstractTable> plus:
 
 =cut
 
-=head2 getData(arguments)
+=head2 getTableRowCount({formData=>{}})
 
-Receive current data for plug-in screen content.
+is not used for card plugins.
 
 =cut
 
-sub getData {
-    my $self = shift;
-    my $call = shift;
-
-        state $valid = {
-        allFields    => 1,
-        deleteEntry  => 1,
-        updateEntry  => 1,
-    };
-    die mkerror(38948,"Unknown sub method $call\n") unless exists $valid->{$call};
-    return $self->$call(@_);
+sub getTableRowCount {
+    croak "Card plugins always get all data and have no getTableRowCount() method."
```

Comment:
this can lead to a denial of service attack as by calling this function an external actor could crash the application ... I would rather just return undef

Reply:
Done.